### PR TITLE
Add --verify Flag Functionality and Tests

### PR DESCRIPTION
**Pull request by rvbcia**
### Summary
This pull request introduces the `--verify` flag functionality to the Helmsman project. The `--verify` flag is used to verify downloaded charts during the execution of the application.

### Changes
1. **Code Changes:**
   - Added the `--verify` flag to the CLI options.
   - Implemented the logic to set and use the `--verify` flag in the main application flow.
   - Updated `internal/app/main.go` to include the `SetVerifyFlag(flags.verify)` function call.

2. **Tests:**
   - Added tests for the `--verify` flag functionality in `internal/app/cli_test.go`.
   - The tests include positive, negative, and edge cases to ensure proper functionality.

3. **Documentation:**
   - Updated the `docs/cmd_reference.md` file to include the `--verify` flag in the list of available CMD options.

### Verification
- Ran unit tests to ensure the new functionality works as expected.
- Verified that the `--verify` flag correctly verifies downloaded charts and handles various scenarios.

### Related Issues
- Reference any related issues or discussions here (if applicable).

### Additional Notes
- Please review the changes and provide feedback.
- Ensure to run the tests locally to verify the functionality before merging.

### CODE REVIEW

**Comment by luisdavim** [comment]
Thanks for the update, can you run `go fmt` ? The formatting seems off.

**Comment by luisdavim** [comment]
> `go fmt` done

Something's off,  I still see some whitepace chagnes that shouldn't be there after running fmt, the branch also needs to be rebased from main and there are some conflicts.

File: internal/app/main.go [hunk 1]
```diff
@@ -37,6 +37,9 @@ func Main() int {
 
 	flags.parse()
 
+	// Set the verify flag in helm_helpers
+	SetVerifyFlag(flags.verify)
```
Comment (luisdavim) [inline internal/app/main.go:41]: why not just follow the same pattern as for the rest of the flags?

File: internal/app/helm_helpers.go [hunk 2]
```diff
@@ -13,25 +13,36 @@ import (
 	"github.com/Praqma/helmsman/internal/gcs"
 )
 
-type helmRepo struct {
-	Name string `json:"name"`
-	URL  string `json:"url"`
+type HelmHelper struct {
```
Comment (luisdavim) [inline internal/app/helm_helpers.go:16]: why is this needed? Seems a bit overkill

**Review by luisdavim — APPROVED** [review]
LGTM, thanks.
